File: solutions/go/rna-transcription/1/rna_transcription.go
package strand

func ToRNA(dna string) string {    
	var rna string
    for _, dnaR := range dna {
        switch dnaR {
        case 'G':
        	rna += string('C')
        case 'C':
            rna += string('G')
        case 'T': 
            rna += string('A')
        case 'A':
            rna += string('U')
        }
    }
    return rna
}

File: solutions/csharp/rna-transcription/1/RnaTranscription.cs
public static class RnaTranscription
{
    public static string ToRna(string strand) => new([.. strand.ToRnaArray()]);

    private static IEnumerable<char> ToRnaArray(this string strand)
    {
        foreach (var c in strand)
        {
            switch (c)
            {
                case 'G':
                    yield return 'C';
                    break;
                case 'C':
                    yield return 'G';
                    break;
                case 'T':
                    yield return 'A';
                    break;
                case 'A':
                    yield return 'U';
                    break;
                default:
                    throw new ArgumentException("Invalid input DNA.");
            }
        }
    }
}